File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.rb
# @param {Integer[][]} matrix
# @return {Void} Do not return anything, modify matrix in-place instead.
def set_zeroes(matrix)
    rows, cols = Set.new, Set.new
    matrix.each_with_index do |row, r|
        row.each_with_index do |cell, c|
            if cell == 0
                rows.add(r)
                cols.add(c)
            end
        end
    end
    rows.each do |r|
        (0...matrix[0].length).each do |c|
            matrix[r][c] = 0
        end
    end
    cols.each do |c|
        (0...matrix.length).each do |r|
            matrix[r][c] = 0
        end
    end
end

File: 0073-set-matrix-zeroes/0073-set-matrix-zeroes.py
class Solution:
    def setZeroes(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        rows, cols = set(), set()
        for r, row in enumerate(matrix):
            for c, cell in enumerate(row):
                if cell == 0:
                    rows.add(r)
                    cols.add(c)
        
        for r in list(rows):
            for c in range(len(matrix[0])):
                matrix[r][c] = 0
        for c in list(cols):
            for r in range(len(matrix)):
                matrix[r][c] = 0